frontend/public/components/dashboards-page/with-dashboard-resources.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import { Map as ImmutableMap } from 'immutable';
+
+import { RESULTS_TYPE } from '../../reducers/dashboards';
+import {
+  watchURL,
+  stopWatchURL,
+  watchPrometheusQuery,
+  stopWatchPrometheusQuery,
+  Fetch,
+  WatchURLAction,
+  WatchPrometheusQueryAction,
+  StopWatchURLAction,
+  StopWatchPrometheusAction,
+} from '../../actions/dashboards';
+import { RootState } from '../../redux';
+
+const mapDispatchToProps = dispatch => ({
+  watchURL: (url, fetch): WatchURL => dispatch(watchURL(url, fetch)),
+  stopWatchURL: (url): StopWatchURL => dispatch(stopWatchURL(url)),
+  watchPrometheusQuery: (query): WatchPrometheus => dispatch(watchPrometheusQuery(query)),
+  stopWatchPrometheusQuery: (query): StopWatchPrometheus => dispatch(stopWatchPrometheusQuery(query)),
+});
+
+const mapStateToProps = (state: RootState) => ({
+  [RESULTS_TYPE.URL]: state.dashboards.get(RESULTS_TYPE.URL),
+  [RESULTS_TYPE.PROMETHEUS]: state.dashboards.get(RESULTS_TYPE.PROMETHEUS),
+});
+
+const WithDashboardResources = (WrappedComponent: React.ComponentType<any>) =>
```

Comment:
`any` will not check that `WrappedComponent` accepts the props we're passing.

Reply:
added more specific type for `WrappedComponent`